### PR TITLE
Feature=latest fixes

### LINKED ISSUE #2092: Embed: `?feature=latest` causing infinite set-states
- From [sen_22570](https://sentry.lbry.tech/organizations/odysee/issues/22570)
    - The `setIsFetching` stack trace might be a red-herring.
    - **In the breadcrumbs, a lot of `user/referral` is being sent.  Maybe this is a better clue.**
- One example of the embed is here (bottom of page): https://www.oval.media/aaf105c4-de54-4643-9a0b-1efdf8aafaad/
    - or this `<iframe id="odysee-iframe" width="560" height="315" src="https://odysee.com/$/embed/@ovalmedia?feature=latest" allowfullscreen></iframe>`
- Locally, the max I've seen is 4 API calls, so still missing the recipe to make it infinite.  The log says it's mostly on mobile .. haven't tried that.

### LINKED ISSUE #2181: embed urls should use canonical url
Right now they use vanity

Check shareurls logic, or just pass canonical.

### PR DESCRIPTION
**Pull request by saltrafael**
## Fixes

Issue Number: closes #2181, closes #2092 (?)

@infinite-persistence please help make sure the `generateEmbedUrlEncoded` still works as intended

attempted to fix 2092 by preventing referral calls if pending or already called, and removing the unmount condition from wrapper component, but not certain if those were causes to infinite loops

_live on salt.odysee.com_

### CODE REVIEW

**Comment by tzarebczan** [comment]
It wasn't only the referrer, it was also calling user/me multiple times too. I can help test this out also.

**Comment by tzarebczan** [comment]
We used to encode characters like @ in the embed URL, but not anymore. Not sure if it needs it or not, but maybe safer? 

Looks like the duplicate call issues are fixed though. Will probably ship this for now cause there's a stream with the live embed tonight.

**Comment by tzarebczan** [comment]
actually, a link like: https://salt.odysee.com/@RT:fd/antiroyal:2 doesn't call referral API anymore. A livestream worked, and channel page worked.

File: ui/redux/actions/user.js [hunk 1]
```diff
@@ -731,82 +739,65 @@ export function doUserSetReferrerWithUri(uri) {
...
+        .catch((error) => dispatch({ type: ACTIONS.USER_SET_REFERRER_FAIL, data: { error } }));
     }
 
-    try {
-      await Lbryio.call('user', 'referral', { referrer: referrerCode }, 'post');
-      dispatch({
-        type: ACTIONS.USER_SET_REFERRER_SUCCESS,
-      });
-      if (shouldClaim) {
+    referrerCode = getChannelFromClaim(claim).permanent_url.replace('lbry://', '');
```
Comment (infinite-persistence) [inline ui/redux/actions/user.js:784]: @saltrafael  This is failing in some cases

https://sentry.lbry.tech/organizations/odysee/issues/30368/?project=2

**Pull request merged**